dataprofiler/profilers/profiler_options.py

```diff
@@ -713,7 +714,13 @@ def _validate_helper(self, variable_path='CategoricalOptions'):
         :return: list of errors (if raise_error is false)
         :rtype: list(str)
         """
-        return super()._validate_helper(variable_path)
+        errors = super()._validate_helper(variable_path)
+        if not (self.top_k_categories is None or (
+                isinstance(self.top_k_categories, int) and self.top_k_categories >= 0)):
+            errors.append("top_k_categories must be either None"
+                          " or a positive integer".format(variable_path))
```

Comment:
I think this would fail would not catch the case where a user could set it to a value which is not an integer.

Reply:
whereas:
```python
if self.top_k_categories is not None and (
                isinstance(self.top_k_categories, int) 
                or self.top_k_categories < 1)):
```
would cover the case when it is not none and it isn't a positive integer

Reply:
Also, we need to utilize the variable_path in the error to that the user knows where this option exists:
```python
errors.append("{}.top_k_categories must be either None"
                          " or a positive integer".format(variable_path))
```
```